### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by DawnSpark**

### CODE REVIEW

File: frontend/src/components/adminPage/ProfessionRows.js [hunk 1]
```diff
@@ -56,19 +71,28 @@ import '../../style/ProfessionRows.css';
...
                 || this.props.idProf!==nextProps.idProf 
                 || this.state.idProf!==nextProps.idProf 
-                || 
-                this.state.currentDate!==nextState.currentDate
+                || this.state.currentDate!==nextState.currentDate
                 || this.state.idProf!==nextState.idProf
                 || this.state.idDoc!==nextState.idDoc
-                || this.state.ratesArr!==nextState.ratesArr);
+                || this.state.ratesArr!==nextState.ratesArr
+                || this.state.shouldUpdate!==nextState.shouldUpdate);
```
Comment (olegRakus) [inline frontend/src/components/adminPage/ProfessionRows.js]: What difference between this record and (this.state !== nextState)?

File: frontend/src/components/adminPage/ProfessionRows.js [hunk 2]
```diff
@@ -56,19 +71,28 @@ import '../../style/ProfessionRows.css';
...
         componentWillUpdate(nextProps, nextState){
             let url_string = window.location.href;
             let url = new URL(url_string);
             let idDoc = url.searchParams.get("doc");
+            let idProf= url.searchParams.get("prof");
+            if(this.state.ratesArr.length!==0)
+            {
+                let currentRate = this.state.ratesArr.find(rate => rate.State==0);
+                var idForDiv = "salaryinfo" + currentRate.StartDate;
+                //$(idForDiv).addClass("active");
```
Comment (olegRakus) [inline frontend/src/components/adminPage/ProfessionRows.js]: please delete commented code

File: frontend/src/components/adminPage/ProfessionRows.js [hunk 3]
```diff
@@ -89,63 +113,84 @@ import '../../style/ProfessionRows.css';
...
+                            <i id="edit" className="fa fa-pencil-alt ml-2" data-toggle="modal" data-target="#EditRateToProfession" onClick = {() => this.changeCurrentDate(rate.StartDate)}></i>
+                            <i id="delete" className="fa fa-times ml-2" data-toggle="modal" data-target="#DeleteRateToProfession" onClick = {() => this.changeCurrentDate(rate.StartDate)}></i>  
                         </div>
                         : (rate.State==0?
-                            images =
-                            <div className="col-3 mt-2" >
-                                <i className="fa fa-pencil-alt col-1 ml-4" data-toggle="modal" data-target="#EditRateToProfession" onClick = {() => this.changeCurrentDate(rate.StartDate)}></i>
+                            (images =
+                            <div className=" row mt-2" >
+                                <i id="edit" className="fa fa-pencil-alt ml-2" data-toggle="modal" data-target="#EditRateToProfession" onClick = {() => this.changeCurrentDate(rate.StartDate)}></i>
```
Comment (olegRakus) [inline frontend/src/components/adminPage/ProfessionRows.js]: Each page can have only one element with that "id", for multiple elements use "class"

File: frontend/src/components/adminPage/modaldialogs/AddRateToProfession.js [hunk 4]
```diff
@@ -31,71 +40,102 @@ class AddRateToProfession extends Component{
...
+            var newCoeff = {
+                DoctorId: Doctor,
+                Coeff: this.rate,
+                StartDate: this.state.startDate.format('YYYY-MM-DD')
+                }   
+                axios.post(localStorage.getItem("server_url") + '/api/Salary/Coefficient/add', newCoeff)
+                .then(response=>{
+                console.log(response.data);
+                this.props.callback(response.data);
+                //$('#DeleteRateToProfession').show('modal');
```
Comment (olegRakus) [inline frontend/src/components/adminPage/modaldialogs/AddRateToProfession.js]: please delete commented code

File: frontend/src/components/adminPage/modaldialogs/EditRateToProfession.js [hunk 5]
```diff
@@ -50,20 +55,25 @@ class EditRateToProfession extends Component{
...
                 } 
                 axios.post(localStorage.getItem("server_url") + '/api/Salary/Coefficient/edit', newCoefficient)
                 .then(response=>{
                 console.log(response.data);
+                this.props.callback(response.data);
                 })
             }
         }
+        // document.getElementById("Rate").textContent="";
+        // this.rate="";
```
Comment (olegRakus) [inline frontend/src/components/adminPage/modaldialogs/EditRateToProfession.js]: please delete commented code

File: frontend/src/components/adminPage/ProfessionRows.js (hunk 1 above)
Comment (IvanovVadym) [reply frontend/src/components/adminPage/ProfessionRows.js]: nothing

Comment (olegRakus) [reply frontend/src/components/adminPage/ProfessionRows.js]: why don't write (this.state !== nextState)?

File: frontend/src/components/adminPage/modaldialogs/EditRateToProfession.js [hunk 6]
```diff
@@ -25,17 +27,20 @@ class EditRateToProfession extends Component{
     }
     
     handleSubmitEdit() {
+        //$("#EditRateToProfession").removeData('modal');
```
Comment (olegRakus) [inline frontend/src/components/adminPage/modaldialogs/EditRateToProfession.js]: please delete commented code

File: frontend/src/components/doctorPage/DoctorCalendar.js [hunk 7]
```diff
@@ -77,27 +80,12 @@ class DoctorCalendar extends React.Component{
...
       eventClick: function(event, jsEvent, view ) {
-        if (event.selectable) {
-          _that.saveCurrentTimeStartEnd(event.start._i, event.end._i);  
+        if (event.selectable) {          
           $("#modButton").trigger("click");
         } else {
-          _that.setId(event.title);
+          _that.saveCurrentTimeStartEnd(event.start._i, event.end._i); 
+          _that.setId(event.patientId, event.patientName, event.start);
           //this.setState({idPatient : event.idPatient});
```
Comment (olegRakus) [inline frontend/src/components/doctorPage/DoctorCalendar.js]: please delete commented code

File: frontend/src/components/doctorPage/PatientMedicalCard/PatientMedicalCard.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Component } from 'react';
+//import PatientDiagnosesTable from '.../PatientDiagnosesTable'
+//import PatientInfo from '../PatientInfo';
```
Comment (olegRakus) [inline frontend/src/components/doctorPage/PatientMedicalCard/PatientMedicalCard.js]: please delete commented code

File: frontend/src/components/doctorPage/PatientMedicalCard/SelectAllergy.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            ]
+        }
+    }
+
+    handleChange = (selectedOption) => {
+        if(selectedOption!==null){
+            this.setState({ selectedOption });
+            //searchParameter.set('allergy', selectedOption.value);
+            // //searchParameter.delete('doc');
+            // window.history.pushState(null, null, `${window.location.pathname}?${searchParameter.toString()}${window.location.hash}`);
```
Comment (olegRakus) [inline frontend/src/components/doctorPage/PatientMedicalCard/SelectAllergy.js]: please delete commented code

File: frontend/src/components/doctorPage/modaldialogs/InfoSchedule.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            this.setState({
+                statusCode: res.data
+            }); 
+        if (this.state.statusCode === 0) {
+            window.location.href = window.location.origin+'/user/medicalCard/'+this.state.idPatient;
+        }
+        else {
+            this.props.callback(this.state.statusCode);
+            // let myColor = { background: '#FF0000', text: "#FFFFFF" };
+            // notify.show("you can not add multiple rates/salaries for one day", "custom", 5000, myColor);
```
Comment (olegRakus) [inline frontend/src/components/doctorPage/modaldialogs/InfoSchedule.js]: please delete commented code

File: frontend/src/components/adminPage/ProfessionRows.js [hunk 11]
```diff
@@ -3,13 +3,14 @@ import { Component } from 'react';
...
+import AddRateToProfession from './modaldialogs/AddRateToProfession'
 import EditRateToProfession from './modaldialogs/EditRateToProfession';
 import DeleteRateToProfession from './modaldialogs/DeleteRateToProfession';
 import 'react-datepicker/dist/react-datepicker.css';
 import validator from 'validator';
+import Notifications, {notify} from 'react-notify-toast';
 import '../../style/ProfessionRows.css';
-
+import $ from 'jquery';
 //import '../style/SalaryReport.css';
```
Comment (olegRakus) [inline frontend/src/components/adminPage/ProfessionRows.js]: please delete commented code

File: frontend/src/components/adminPage/AdminSwitch.js [hunk 12]
```diff
@@ -19,7 +19,7 @@ class AdminSwitch extends Component {
             <Route exact path="/admin/salary" component={Edit}/> */}
```
Comment (olegRakus) [inline frontend/src/components/adminPage/AdminSwitch.js]: please delete commented code

**Pull request merged**